File: Base Binding Classes/MainFragment.java
package com.example.myapplication;

import android.view.View;
import com.example.myapplication.databinding.FragmentMainBinding;

public class MainFragment extends BaseFragment<FragmentMainBinding> {

    private static final String TAG = "MainFragment";
    private static final String TEXT = "Hello World!";

    @Override
    protected int getLayoutResID() {
        return R.layout.fragment_main;
    }

    @Override
    protected void initViews(View view) {
        binding.tvText.setText(TEXT);
    }
}

File: Base Binding Classes/MainFragment.kt
package com.example.myapplication

import android.view.View
import com.example.myapplication.databinding.FragmentMainBinding

class MainFragment: BaseFragment<FragmentMainBinding>(R.layout.fragment_main) {

    companion object{
        const val TAG = "MainFragment"
        const val TEXT = "Hello World!"
    }

    override fun initViews(view: View) {
        /*
         *  UI created
         *  your code
         */
        binding?.tvText?.text = TEXT
    }
}
